src/wallet/wallet.cpp

```diff
@@ -1077,7 +1081,14 @@ void CopyPreviousWitnesses(NoteDataMap& noteDataMap, int indexHeight, int64_t nW
             assert(nWitnessCacheSize >= nd->witnesses.size());
             // Witnesses being incremented should always be either -1
             // (never incremented or decremented) or one below indexHeight
-            assert((nd->witnessHeight == -1) || (nd->witnessHeight == indexHeight - 1));
+            assert((nd->witnessHeight == -1) || (nd->witnessHeight <= indexHeight - 1));
+            if (nd->witnessHeight != -1 && nd->witnessHeight < indexHeight-1) {
+                LogPrintf("Unexpected note witness height, clearing witnesses "
+                        "txid=%s indexHeight=%d ndHeight=%d\n",
+                    txhash.ToString(), indexHeight, nd->witnessHeight);
+                nd->witnesses.clear();
```

Comment:
This is the same fix as below, https://github.com/zcash/zcash/pull/4573/files#diff-b2bb174788c7409b671c46ccc86034bdR1137 in response to the same condition.